### PR TITLE
Game start scene

### PR DESCRIPTION
**Pull request by joefinn01**

### CODE REVIEW

File: CoolEngine/CoolEngine.vcxproj [hunk 1]
```diff
@@ -628,7 +628,7 @@
       <WarningLevel>Level3</WarningLevel>
       <FunctionLevelLinking>true</FunctionLevelLinking>
       <IntrinsicFunctions>true</IntrinsicFunctions>
-      <SDLCheck>true</SDLCheck>
+      <SDLCheck>false</SDLCheck>
```
Comment (Abhiroop-Basak) [inline CoolEngine/CoolEngine.vcxproj:631]: What is this change for ? Just making sure you are aware.

File: CoolEngine/Engine/Physics/Collision.cpp [hunk 2]
```diff
@@ -345,14 +348,20 @@ void Collision::Update(vector<GameObject*>& gameObjectMap, vector<Tile*>& tiles)
 			{
 				XMFLOAT3 p = pcollidable->GetTransform()->GetWorldPosition();
 				XMFLOAT2 d = pcollidable->GetShape()->GetShapeDimensions();
+
+#if _DEBUG
```
Comment (Abhiroop-Basak) [inline CoolEngine/Engine/Physics/Collision.cpp:352]: Does _DEBUG and EDITOR have different funtionalities ?

File: CoolEngine/Engine/Physics/Collision.cpp [hunk 3]
```diff
@@ -345,14 +348,20 @@ void Collision::Update(vector<GameObject*>& gameObjectMap, vector<Tile*>& tiles)
...
 				DebugDrawManager::GetInstance()->CreateWorldSpaceDebugRect(p, XMFLOAT3(d.x, d.y, 1), DebugDrawManager::DebugColour::PURPLE);
+#endif
 			}
 
 			if (tiles[j]->GetShape() != nullptr && tiles[j]->GetShape()->IsRendered())
 			{
 				XMFLOAT3 p = tiles[j]->GetTransform()->GetWorldPosition();
 				XMFLOAT2 d = tiles[j]->GetShape()->GetShapeDimensions();
+
+#if _DEBUG
```
Comment (Abhiroop-Basak) [inline CoolEngine/Engine/Physics/Collision.cpp:362]: Same as above

**Review by Abhiroop-Basak — CHANGES_REQUESTED** [review]

File: CoolEngine/CoolEngine.vcxproj (hunk 1 above)
Comment (joefinn01) [reply CoolEngine/CoolEngine.vcxproj:631]: It was some weird compilation errors where it was saying some pointers could be uninitialized but they are.

File: CoolEngine/Engine/Physics/Collision.cpp (hunk 2 above)
Comment (joefinn01) [reply CoolEngine/Engine/Physics/Collision.cpp:352]: Yes on is for the debug configurations and the other is for editor and game

File: CoolEngine/Engine/Physics/Collision.cpp (hunk 3 above)
Comment (joefinn01) [reply CoolEngine/Engine/Physics/Collision.cpp:362]: Same as above

**Review by Abhiroop-Basak — APPROVED** [review]

**Pull request merged**